### ISSUE TITLE
Detach Servo with controlled stop of waveform

### ISSUE THREAD

**Issue by obrain17** [esp8266/Arduino#7138]
The current code from master detaches a servo within a running PWM duty cycle. This can lead to twitches and in worst case destroy the servo.

There are already comments and PRs for solving this issue. #7084 and  #7023
This issue is opened to follow up with this PRs.

**Comment by Tech-TX** [comment]
This was corrected with #7022 and #7231.   A quick test with 2 servos shows that it always stops at the last position before the servo.detach(), and never hits the stops.  I attached, swept and detached the servos 40-50 times, no wild sweeps to the stops at detach.
